relayer/processor/path_processor_internal.go

```diff
@@ -244,7 +236,29 @@ func (pp *PathProcessor) sendMessages(
 		messages = append(messages, msg.message)
 	}
 
-	_, txSuccess, err := dst.chainProvider.SendMessages(ctx, messages)
+	if len(messages) == 0 {
+		pp.log.Debug("No messages to send after filtering", zap.String("chain_id", dst.info.ChainID))
+		return nil
+	}
+
+	msgUpdateClient, err := pp.assembleMsgUpdateClient(ctx, src, dst)
+	if err != nil {
+		pp.log.Debug("Error prepending MsgUpdateClient",
```

Comment:
Another potential double-logged error, but I mind less for a debug-level log.

Reply:
This log is part of the `sendMessages` function, which is called in an error group, so I wanted to avoid swallowing errors if there are errors from both sides. But to avoid double logging, I changed this to return a wrapped error here and moved logging into the errgroup goroutine callers of `sendMessages`.